src/components/encounter/EncounterWelcome.tsx

```diff
@@ -0,0 +1,142 @@
+import { NavigationProp } from '@react-navigation/core';
+import React, { useCallback, useContext, useMemo, useState } from 'react';
+import { useQuery } from 'react-apollo';
+import { ScrollView, StyleSheet, View } from 'react-native';
+
+import {
+  Button,
+  Image,
+  LoadingSpinner,
+  RegularText,
+  SafeAreaViewFlex,
+  SectionHeader,
+  Wrapper,
+  WrapperWithOrientation
+} from '..';
+import { texts } from '../../config';
+import { graphqlFetchPolicy } from '../../helpers';
+import { usePullToRefetch, useRefreshTime } from '../../hooks';
+import { parseEncounterWelcome } from '../../jsonValidation';
+import { NetworkContext } from '../../NetworkProvider';
+import { getQuery, QUERY_TYPES } from '../../queries';
+import { ScreenName } from '../../types';
+
+type StaticContentArgs<T = unknown> = {
+  publicJsonFile: string;
+  refreshTimeKey?: string;
+  refreshInterval?: string;
+  parseJson?: (json: unknown) => T;
+};
+
+const useStaticContent = <T,>({
+  publicJsonFile,
+  parseJson,
+  refreshInterval,
+  refreshTimeKey
+}: StaticContentArgs<T>): {
+  data: T;
+  error: boolean;
+  loading: boolean;
+  refetch: () => Promise<unknown>;
+} => {
+  const { isConnected, isMainserverUp } = useContext(NetworkContext);
+  const [error, setError] = useState(false);
+
+  const refreshTime = useRefreshTime(refreshTimeKey ?? publicJsonFile, refreshInterval);
+
+  const fetchPolicy = graphqlFetchPolicy({
+    isConnected,
+    isMainserverUp,
+    refreshTime
+  });
+
+  const { data, error: queryError, loading, refetch } = useQuery(
+    getQuery(QUERY_TYPES.PUBLIC_JSON_FILE),
+    {
+      variables: { name: publicJsonFile },
+      fetchPolicy,
+      skip: !refreshTime
+    }
+  );
+
+  const refetchCallback = useCallback(async () => {
+    setError(false);
+    return await refetch?.();
+  }, [refetch]);
+
+  const publicJsonFileData = useMemo(() => {
+    try {
+      if (data) {
+        const json = JSON.parse(data?.publicJsonFile?.content);
+        return parseJson ? parseJson(json) : json;
+      }
+    } catch (error) {
+      setError(true);
+      console.warn(error, data);
+    }
+  }, [data, parseJson]);
+
+  return {
+    data: publicJsonFileData,
+    error: !!(queryError || error),
+    loading: loading || (!publicJsonFileData && !error),
```

Comment:
reading the second line i like the way that `loading: loading ...` marks the "extended" loading condition. the first line is the other way around for no functional reason. let's do `error: !!(error || queryError)` as well to point out, that we pass through the value with additional condition?